apps/src/lib/cli.rs

```diff
@@ -697,43 +702,53 @@ pub mod args {
     /// Craft intent for token exchange arguments
     #[derive(Debug)]
     pub struct CraftIntent {
+        pub key: Address,
         /// Source address
-        pub addr: Address,
+        pub addr: Vec<Address>,
         /// Token to sell
-        pub token_sell: Address,
-        /// Amount of token to sell
-        pub amount_sell: token::Amount,
+        pub token_sell: Vec<Address>,
+        /// Max amount of token to sell
+        pub max_sell: Vec<token::Amount>,
+        /// Rate
+        pub min_rate: Vec<DecimalWrapper>,
         /// Token to buy
-        pub token_buy: Address,
-        /// Amount of token to buy
-        pub amount_buy: token::Amount,
+        pub token_buy: Vec<Address>,
+        /// Min amount of token to buy
+        pub min_buy: Vec<token::Amount>,
```

Comment:
I think since all these `Vec` must have the same length, it would be nicer to make the data type e.g.:

```
    pub struct CraftIntent {
        /// A key that will sign the whole intent
        pub key: Address,
        pub exchanges: Vec<TokenExchange>,
        /// Target file path
        pub file_path: String,
    }

    pub struct TokenExchange {
        /// Source address that will sign the exchange
        pub addr: Address,
        /// Token to sell
        pub token_sell: Address,
        /// Max amount of token to sell
        pub max_sell: token::Amount,
        /// Rate
        pub min_rate: DecimalWrapper,
        /// Token to buy
        pub token_buy: Address,
        /// Min amount of token to buy
        pub min_buy: token::Amount,
    }
```